client/src/app/employee/employee-list/employee-list.component.ts

```diff
@@ -20,6 +21,8 @@ export class EmployeeListComponent implements OnInit, OnDestroy {
   constructor(
     private service: EmployeeListService,
     private query: EmployeeListQuery,
+    private store: EmployeeListStore,
```

Comment:
The store should only be accessible by the Query and the Service, nothing else.

This is what forces an architecture of 1 way data flow.

Changes go to the -> Service

Service changes the -> Store

Store emits which is exposed by the -> Query

Query passes change/data streams to the -> Component

Component passes a requested state change to the -> Service ... ect

Reply:
fixed according to our discussion this morning!